sdks/python/apache_beam/runners/portability/fn_api_runner.py

```diff
@@ -810,10 +907,12 @@ def input_for(transform_id, input_id):
     return result
 
   @staticmethod
-  def _extract_endpoints(stage,
-                         pipeline_components,
-                         data_api_service_descriptor,
-                         pcoll_buffers):
+  def _extract_endpoints(stage,  # type: fn_api_runner_transforms.Stage
+                         pipeline_components,  # type: beam_runner_api_pb2.Components
+                         data_api_service_descriptor, # type: Optional[endpoints_pb2.ApiServiceDescriptor]
+                         pcoll_buffers  # type: DefaultDict[bytes, _ListBuffer]
+                        ):
+    # type: (...) -> Tuple[Dict[str, _ListBuffer], Dict[Tuple[str, str], Tuple[bytes, beam_runner_api_pb2.FunctionSpec]], Dict[str, bytes]]
```

Comment:
Tuple[Dict[str, _ListBuffer], DataSideInput, DataOutput]?

Reply:
yep, good catch.  this was accidentally left in the second set of commits (post-part1). I'll go ahead and move that into this PR.